hazelcast/src/main/java/com/hazelcast/config/DynamicConfigGenerator.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.internal.util.XmlUtil;
+import com.hazelcast.spi.annotation.PrivateApi;
+import org.snakeyaml.engine.v2.api.Dump;
+import org.snakeyaml.engine.v2.api.DumpSettings;
+import org.snakeyaml.engine.v2.common.FlowStyle;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+@PrivateApi
+public final class DynamicConfigGenerator {
```

Comment:
Maybe move to `com.hazelcast.internal.config` package?

Reply:
I moved into `com.hazelcast.internal.config.dynamic`. I think this makes more sense as it's related to dynamic config. And also in the future I will also move everything from `com.hazelcast.internal.dynamicconfig` to `com.hazelcast.internal.config.dynamic`. I think it makes more sense. I believe the reason it's not like this currently is because `com.hazelcast.internal.dynamicconfig` created before `com.hazelcast.internal.config`.